lib/Dialect/OM/OMOps.cpp

```diff
@@ -336,60 +336,7 @@ circt::om::ObjectOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
 
 LogicalResult
 circt::om::ObjectFieldOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
-  // Get the ObjectInstOp and the ClassLike it is an instance of.
-  ObjectOp objectInst = getObject().getDefiningOp<ObjectOp>();
-  ClassLike classDef = cast<ClassLike>(symbolTable.lookupNearestSymbolFrom(
-      *this, objectInst.getClassNameAttr()));
-
-  // Traverse the field path, verifying each field exists.
-  ClassFieldLike finalField;
-  auto fields = SmallVector<FlatSymbolRefAttr>(
-      getFieldPath().getAsRange<FlatSymbolRefAttr>());
-  for (size_t i = 0, e = fields.size(); i < e; ++i) {
-    // Verify the field exists on the ClassOp.
-    auto field = fields[i];
-    ClassFieldLike fieldDef;
-    classDef.walk([&](ClassFieldLike fieldLike) {
-      if (fieldLike.getNameAttr() == field.getAttr()) {
-        fieldDef = fieldLike;
-        return WalkResult::interrupt();
-      }
-      return WalkResult::advance();
-    });
-    if (!fieldDef) {
-      auto error = emitOpError("referenced non-existant field ") << field;
-      error.attachNote(classDef.getLoc()) << "class defined here";
-      return error;
-    }
-
-    // If there are more fields, verify the current field is of ClassType, and
-    // look up the ClassOp for that field.
-    if (i < e - 1) {
-      auto classType = dyn_cast<ClassType>(fieldDef.getType());
-      if (!classType)
-        return emitOpError("nested field access into ")
-               << field << " requires a ClassType, but found "
-               << fieldDef.getType();
-
-      // The nested ClassOp must exist, since a field with ClassType must be
-      // an ObjectInstOp, which already verifies the class exists.
-      classDef = cast<ClassLike>(
-          symbolTable.lookupNearestSymbolFrom(*this, classType.getClassName()));
-
-      // Proceed to the next field in the path.
-      continue;
-    }
-
-    // On the last iteration down the path, save the final field being accessed.
-    finalField = fieldDef;
-  }
-
-  // Verify the accessed field type matches the result type.
-  if (finalField.getType() != getResult().getType())
-    return emitOpError("expected type ")
-           << getResult().getType() << ", but accessed field has type "
-           << finalField.getType();
-
+  // This is verified by VerifyObjectFields pass.
```

Comment:
Is there utility in implementing this / SymbolUserOpInterface if it is only ever success (and our symbol users probably don't work right anyway -- they aren't underneath the SymbolTable defining the symbols referenced AFAICT).

Anyway maybe fine for now, pending a rework esp now we're managing our own verification too.

Reply:
I think when we finally fix this for real, this will probably be where we re-land the verifier, since `om.object.field` will still reference the `om.class` symbol. The difference is we will re-work the IR to not have to then scan the `om.class` body to perform the verification. So I guess this is fine to rip out now, and we can add it back then, or we can leave the placeholder.

Reply:
I was thinking that since these aren't not compatible with MLIR symbols the rework wouldn't use this since these aren't symbol users in the normal sense?

But okay, something to chat about later...

Reply:
Yeah I think we can drop the interface. ObjectFieldOp would not be a MLIR symbol user.